### DISCUSSION TITLE
Check workflow status from within a step using actions/github-script

### DISCUSSION THREAD

**Discussion by pantelis-karamolegkos** [community/community#109684] [Actions] [answered]
### Select Topic Area

Question

### Body

I am using the action [script][1] 

I want to check from within the step that uses it it whether previous steps were successful however the following line is marked with error

```
"color": "${{ success() && '#36a64f' || '#ecb22e' }}",
```

Is there a way to go about it? It goes without saying that the specific step has the `if: always()` condition.

  [1]: https://github.com/actions/github-script

**Comment by thisisthetechie** [comment]
Off the top of my head it's because `success()` might not be available within a step - you would probably need to give it more information (such as `steps.mystep.outcome` or `steps.mystep.conclusion` perhaps?)

**Reply by pantelis-karamolegkos** [reply]
Yes but I want to check whether any of the previous steps has failed (not a specific one) i.e. I want to reproduce the `if: success()` condition of a step execution from within the script

**Reply by thisisthetechie** [reply]
Okay. `toJson` might help you here:
```
- run: echo "${{ toJson(steps) }}"
```
Would give you the following (assuming your steps have an `id` set - otherwise they don't produce any output):
```
{
  my-step-1: {
    outputs: {},
    outcome: success,
    conclusion: success
  },
  my-step-2: {
    outputs: {},
    outcome: success,
    conclusion: success
  }
}
```

I would even guess that you can do:
```
"color": "${{ contains(steps.*.outcome, 'failure') && '#ecb22e' || '#36a64f' }}",
```

**Reply by thisisthetechie** [reply]
Or you can have 2 steps to do the colouring in for you... One with an `if: ${{ failure() }}` and one with `if: ${{ success() }}`? Just an obvious answer :)

**Reply by pantelis-karamolegkos** [reply, marked as answer]
I solved it like this

```
"color": "${{ job.status == 'success' && '#36a64f' || '#ecb22e' }}",
```
